linux/src/main.c

```diff
@@ -279,6 +281,54 @@ static void main_loop_epoll_add(enum main_loop_epoll_type type, int fd) {
   check_posix(epoll_ctl(config.epollfd, EPOLL_CTL_ADD, fd, &ev), "epoll_ctl");
 }
 
+static void get_peripherals() {
+  FILE *fp;
+  char buffer[64];
+
+  /* Read motors and sensors */
+  fp = popen("set -- address driver_name position speed; for f in /sys/class/tacho-motor/*; do for item in $@; do cat $f/$item; done; done; set -- address driver_name mode value0; for f in /sys/class/lego-sensor/*; do for item in $@; do cat $f/$item; done; done", "r");
```

Comment:
I think it would be better if this were written in C rather than shelling out..

Reply:
Resolved in f8c0474cb555d881b4fb35bfec9ff6c8db152e1e